internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
+	echo.Context
+	AppLogger *zap.Logger
+}
+
+func CreateAppContext(
```

Comment:
In the `CreateAppContext` function, the parameter `logger` should be validated to ensure it's not nil before using it. This prevents potential runtime panics in the case of a nil logger being passed in.